install/azure.go

```diff
@@ -72,16 +85,36 @@ func (k *K8sInstaller) retrieveSubscriptionID(ctx context.Context) error {
 		return fmt.Errorf("unable to unmarshal az output: %w", err)
 	}
 
-	k.Log("✅ Derived Azure Subscription ID %s from subscription %s", ai.ID, ai.Name)
+	k.Log("✅ Derived Azure subscription ID %s from subscription %s", ai.ID, ai.Name)
 	k.params.Azure.SubscriptionID = ai.ID
 
 	return nil
 }
 
-func (k *K8sInstaller) createAzureServicePrincipal(ctx context.Context) error {
+func (k *K8sInstaller) azureRetrieveAksNodeResourceGroup(ctx context.Context) error {
+	args := []string{"aks", "show", "--subscription", k.params.Azure.SubscriptionID, "--resource-group", k.params.Azure.ResourceGroupName, "--name", k.params.ClusterName}
+	cmd := azCommand(args...)
+	bytes, err := cmd.Output()
+	if err != nil {
+		return fmt.Errorf("unable to execute \"az %s\": %w", args, err)
+	}
+
+	clusterInfo := aksClusterInfo{}
+	if err := json.Unmarshal(bytes, &clusterInfo); err != nil {
+		return fmt.Errorf("unable to unmarshal az output: %w", err)
+	}
+
+	k.Log("✅ Derived AKS node resource group %s from resource group %s", clusterInfo.NodeResourceGroup, k.params.Azure.ResourceGroupName)
+	k.params.Azure.NodeResourceGroup = clusterInfo.NodeResourceGroup
+
+	return nil
+}
+
+func (k *K8sInstaller) azureSetupServicePrincipal(ctx context.Context) error {
 	if k.params.Azure.TenantID == "" && k.params.Azure.ClientID == "" && k.params.Azure.ClientSecret == "" {
 		k.Log("🚀 Creating Azure Service Principal for Cilium operator...")
-		args := []string{"ad", "sp", "create-for-rbac", "--scopes", "/subscriptions/" + k.params.Azure.SubscriptionID}
+		// Restrict the scope of the Service Principal to the strict minimum (i.e. the resource group)
+		args := []string{"ad", "sp", "create-for-rbac", "--scopes", "/subscriptions/" + k.params.Azure.SubscriptionID + "/resourceGroups/" + k.params.Azure.NodeResourceGroup}
```

Comment:
I've always wondered what this does behind the scenes. Will this create a new SP on each invocation, or does it re-use an existing one? I'm yet to discover where to view active SPs in the Azure Portal. :sweat_smile:

Reply:
@ti-mo See https://github.com/cilium/cilium-cli/pull/308#discussion_r650312351.

Maybe this is a good indication a comment is needed around here :slightly_smiling_face:

Reply:
As said above, in this instance it will create a new dedicated SP on each call because we did not pass a name to `az ad sp create-for-rbac`. If `--name` is passed, then on first call it creates a new SP with the name, and on subsequent calls it updates the SP with a new client ID and client secret, but keeping the same SP (this is what we used to have before we decided to implement #201, but is definitely not what we want).

Reply:
Documented in latest push.

Reply:
Very solid, thank you :+1: